bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -4568,6 +4568,26 @@ void RewriteInstance::updateELFSymbolTable(
     }
   }
 
+  // Add runtime library start and fini address symbols
+  if (RuntimeLibrary *RtLibrary = BC->getRuntimeLibrary()) {
+    auto AddSymbol = [&](StringRef Name, uint64_t Address) {
```

Comment:
I don't understand why we're defining a new AddSymbol lambda here? Can't we just move the definition that we already have in line 4597 to line 4570 and use that?

Reply:
It is not a principal question, but is there a logic behind marking new inserted symbols as weak? Basically I don't mind to move it upper, but I was curious about the logic on this.

Reply:
Oh I see, these symbols are LOCAL and the other lambda inserts WEAK syms.

I don't think these symbols need to be weak to be honest. But I don't quite remember why they are set as weak. But maybe you can try using the same lambda with one extra parameter:  binding ? So in one callsite you set the parameter to local, and in the other callsite you set it to weak.

Reply:
I don't really think it does matter. What does matter is that I realised why at first I did use __bolt_runtime_start and not __bolt_instr_start, it is because it is not instrumentation-specific code, but runtime and every runtime library would have it's own naming. So I would revert my last change back to old symbol name. That is why I we can't use old lambda also, since we're creating this symbol and didn't emit the symbol with such name previously.

Reply:
I see. LGTM

Reply:
Although I can pass the address as second argument. Probably would do it like so. And weak or local - I don't really think it does matter